pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -553,3 +694,60 @@ func (c *Cluster) resolveNode(nodeName string) *rookalpha.Node {
 
 	return rookNode
 }
+
+func (c *Cluster) prepareStorageClassDeviceSets(config *provisionConfig) []rookalpha.VolumeSource {
+	volumeSources := []rookalpha.VolumeSource{}
+	for _, storageClassDeviceSet := range c.DesiredStorage.StorageClassDeviceSets {
+		if err := opspec.CheckPodMemory(storageClassDeviceSet.Resources, cephOsdPodMinimumMemory); err != nil {
+			config.addError("cannot use storageClassDeviceSet %s for creating osds %v", storageClassDeviceSet.Name, err)
+			continue
+		}
+		for setIndex := 0; setIndex < storageClassDeviceSet.Count; setIndex++ {
```

Comment:
nit: how about `i` for the iteration var?

Reply:
cc @ashishranjan738

Reply:
done